### PR TITLE
Add ignore functions

### PR DESCRIPTION
**Pull request by oyvindrobertsen**

### CODE REVIEW

File: src/egit-ignore.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    git_repository *repo = EGIT_EXTRACT(_repo);
+    int retval;
+    {
+        char *rules = EGIT_EXTRACT_STRING(_rules);
+        retval = git_ignore_add_rule(repo, rules);
+        free(rules);
+    }
+    EGIT_CHECK_ERROR(retval);
+
+    return env->make_integer(env, retval);
```
Comment (TheBB) [inline src/egit-ignore.c]: I usually just returm `em_nil` in cases like this to be honest. Doesn't really matter either way.

File: src/egit-ignore.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    int out;
+    int retval;
+    {
+        char *path = EGIT_EXTRACT_STRING(_path)
+        retval = git_ignore_path_is_ignored(&out, repo, path);
+        free(path);
+    }
+    EGIT_CHECK_ERROR(retval);
+    return env->make_integer(env, out);
```
Comment (TheBB) [inline src/egit-ignore.c]: Predicates should return bools: `return out ? em_t : em_nil`

**Comment by TheBB** [comment]
Oh, and I think your testfile has a missing close paren, that's why the CI is complaining.

**Comment by oyvindrobertsen** [comment]
> Oh, and I think your testfile has a missing close paren, that's why the CI is complaining.

Thanks!

**Pull request merged**